### PR TITLE
Addition of description field for virtual server/pool/node with acceptance tests

### PR DESCRIPTION
**Pull request by RavinderReddyF5**
Fix for below issue with addition of Acceptance test and document update
https://github.com/terraform-providers/terraform-provider-bigip/issues/121

Test Results:
```
root@terraformclient:~/Go_Workspace/src/github.com/terraform-providers/terraform-provider-bigip# BIGIP_HOST=10.145.67.133 BIGIP_USER=admin BIGIP_PASSWORD=F5site02 make testacc
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./... -v  -timeout 120m
?       github.com/terraform-providers/terraform-provider-bigip [no test files]
=== RUN   TestAccProvider
--- PASS: TestAccProvider (0.00s)
=== RUN   TestAccBigipCmDevice_create
--- PASS: TestAccBigipCmDevice_create (0.84s)
=== RUN   TestAccBigipCmDevice_import
--- PASS: TestAccBigipCmDevice_import (0.57s)
=== RUN   TestAccBigipCmDevicegroup_create
--- PASS: TestAccBigipCmDevicegroup_create (0.91s)
=== RUN   TestAccBigipCmDevicegroup_import
--- PASS: TestAccBigipCmDevicegroup_import (2.39s)
=== RUN   TestAccBigipLtmDataGroup_create
--- PASS: TestAccBigipLtmDataGroup_create (1.94s)
=== RUN   TestAccBigipLtmDataGroup_import
--- PASS: TestAccBigipLtmDataGroup_import (2.44s)
=== RUN   TestAccBigipLtmIRule_create
--- PASS: TestAccBigipLtmIRule_create (0.72s)
=== RUN   TestAccBigipLtmIRule_import
--- PASS: TestAccBigipLtmIRule_import (0.93s)
=== RUN   TestAccBigipLtmMonitor_create
--- PASS: TestAccBigipLtmMonitor_create (3.81s)
=== RUN   TestAccBigipLtmMonitor_import
--- PASS: TestAccBigipLtmMonitor_import (1.19s)
=== RUN   TestAccBigipLtmNode_create
--- PASS: TestAccBigipLtmNode_create (1.00s)
=== RUN   TestAccBigipLtmNode_import
--- PASS: TestAccBigipLtmNode_import (0.97s)
=== RUN   TestAccBigipLtmNodeInvalid
--- PASS: TestAccBigipLtmNodeInvalid (0.01s)
=== RUN   TestAccBigipLtmNodeCreate
--- PASS: TestAccBigipLtmNodeCreate (0.09s)
=== RUN   TestAccBigipLtmPersistenceProfileCookieCreate
--- PASS: TestAccBigipLtmPersistenceProfileCookieCreate (0.67s)
=== RUN   TestAccBigipLtmPersistenceProfileCookieImport
--- PASS: TestAccBigipLtmPersistenceProfileCookieImport (0.87s)
=== RUN   TestAccBigipLtmPersistenceProfileDstAddrCreate
--- PASS: TestAccBigipLtmPersistenceProfileDstAddrCreate (0.64s)
=== RUN   TestAccBigipLtmPersistenceProfileDstAddrImport
--- PASS: TestAccBigipLtmPersistenceProfileDstAddrImport (0.81s)
=== RUN   TestAccBigipLtmPersistenceProfileSrcAddrCreate
--- PASS: TestAccBigipLtmPersistenceProfileSrcAddrCreate (0.91s)
=== RUN   TestAccBigipLtmPersistenceProfileSrcAddrImport
--- PASS: TestAccBigipLtmPersistenceProfileSrcAddrImport (0.87s)
=== RUN   TestAccBigipLtmPersistenceProfileSSLCreate
--- PASS: TestAccBigipLtmPersistenceProfileSSLCreate (0.89s)
=== RUN   TestAccBigipLtmPersistenceProfileSSLImport
--- PASS: TestAccBigipLtmPersistenceProfileSSLImport (0.89s)
=== RUN   TestAccBigipLtmPolicy_create
--- PASS: TestAccBigipLtmPolicy_create (1.87s)
=== RUN   TestAccBigipLtmPolicy_import
--- PASS: TestAccBigipLtmPolicy_import (1.83s)
=== RUN   TestAccBigipLtmPool_create
--- PASS: TestAccBigipLtmPool_create (0.81s)
=== RUN   TestAccBigipLtmPool_import
--- PASS: TestAccBigipLtmPool_import (0.80s)
=== RUN   TestAccBigipLtmfasthttp_create
--- PASS: TestAccBigipLtmfasthttp_create (0.58s)
=== RUN   TestAccBigipLtmProfilefasthttp_import
--- PASS: TestAccBigipLtmProfilefasthttp_import (1.10s)
=== RUN   TestAccBigipLtmProfileFastl4_create
--- PASS: TestAccBigipLtmProfileFastl4_create (0.66s)
=== RUN   TestAccBigipLtmProfileFastl4_import
--- PASS: TestAccBigipLtmProfileFastl4_import (0.72s)
=== RUN   TestAccBigipLtmProfileHttp2_create
--- PASS: TestAccBigipLtmProfileHttp2_create (0.47s)
=== RUN   TestAccBigipLtmProfileHttp2_modify
--- PASS: TestAccBigipLtmProfileHttp2_modify (0.86s)
=== RUN   TestAccBigipLtmProfileHttp2_import
--- PASS: TestAccBigipLtmProfileHttp2_import (0.54s)
=== RUN   TestAccBigipLtmProfilehttp_create
--- PASS: TestAccBigipLtmProfilehttp_create (0.91s)
=== RUN   TestAccBigipLtmProfilehttp_import
--- PASS: TestAccBigipLtmProfilehttp_import (0.57s)
=== RUN   TestAccBigipLtmProfileHttpcompress_create
--- PASS: TestAccBigipLtmProfileHttpcompress_create (0.74s)
=== RUN   TestAccBigipLtmProfileHttpcompress_import
--- PASS: TestAccBigipLtmProfileHttpcompress_import (0.92s)
=== RUN   TestAccBigipLtmProfileoneconnect_create
--- PASS: TestAccBigipLtmProfileoneconnect_create (0.50s)
=== RUN   TestAccBigipLtmProfileoneconnect_import
--- PASS: TestAccBigipLtmProfileoneconnect_import (0.49s)
=== RUN   TestAccBigipLtmProfileTcp_create
--- PASS: TestAccBigipLtmProfileTcp_create (0.45s)
=== RUN   TestAccBigipLtmProfileTcp_import
--- PASS: TestAccBigipLtmProfileTcp_import (0.52s)
=== RUN   TestAccBigipLtmsnat_create
--- PASS: TestAccBigipLtmsnat_create (0.54s)
=== RUN   TestAccBigipLtmsnat_import
--- PASS: TestAccBigipLtmsnat_import (0.63s)
=== RUN   TestAccBigipLtmsnatpool_create
--- PASS: TestAccBigipLtmsnatpool_create (0.51s)
=== RUN   TestAccBigipLtmsnatpool_import
--- PASS: TestAccBigipLtmsnatpool_import (0.47s)
=== RUN   TestAccBigipLtmVA_create
--- PASS: TestAccBigipLtmVA_create (0.54s)
=== RUN   TestAccBigipLtmVA_import
--- PASS: TestAccBigipLtmVA_import (0.48s)
=== RUN   TestAccBigipLtmVS_create
--- PASS: TestAccBigipLtmVS_create (1.94s)
=== RUN   TestAccBigipLtmVS_import
--- PASS: TestAccBigipLtmVS_import (1.90s)
=== RUN   TestAccBigipNetroute_create
--- PASS: TestAccBigipNetroute_create (1.17s)
=== RUN   TestAccBigipNetroute_import
--- PASS: TestAccBigipNetroute_import (0.96s)
=== RUN   TestAccBigipNetselfip_create
--- PASS: TestAccBigipNetselfip_create (1.14s)
=== RUN   TestAccBigipNetselfip_import
--- PASS: TestAccBigipNetselfip_import (2.53s)
=== RUN   TestAccBigipNetvlan_create
--- PASS: TestAccBigipNetvlan_create (2.60s)
=== RUN   TestAccBigipNetvlan_import
--- PASS: TestAccBigipNetvlan_import (2.64s)
=== RUN   TestAccBigipSysdns_create
--- PASS: TestAccBigipSysdns_create (0.58s)
=== RUN   TestAccBigipSysdns_import
--- PASS: TestAccBigipSysdns_import (0.52s)
=== RUN   TestAccBigipSysIapp_create
--- PASS: TestAccBigipSysIapp_create (1.43s)
=== RUN   TestAccBigipSysIapp_import
--- PASS: TestAccBigipSysIapp_import (0.86s)
=== RUN   TestAccBigipSysNtp_create
--- PASS: TestAccBigipSysNtp_create (0.56s)
=== RUN   TestAccBigipSysNtp_import
--- PASS: TestAccBigipSysNtp_import (0.52s)
=== RUN   TestAccBigipSysNtpInvalid
--- PASS: TestAccBigipSysNtpInvalid (0.01s)
=== RUN   TestAccBigipSysProvision_create
--- PASS: TestAccBigipSysProvision_create (0.42s)
=== RUN   TestAccBigipSysProvision_import
--- PASS: TestAccBigipSysProvision_import (0.74s)
=== RUN   TestAccBigipSyssnmp_create
--- PASS: TestAccBigipSyssnmp_create (1.19s)
=== RUN   TestAccBigipSyssnmp_import
--- PASS: TestAccBigipSyssnmp_import (0.61s)
=== RUN   TestValidStringValue
--- PASS: TestValidStringValue (0.00s)
=== RUN   TestInvalidStringValue
--- PASS: TestInvalidStringValue (0.00s)
=== RUN   TestF5NameString
--- PASS: TestF5NameString (0.00s)
=== RUN   TestF5NameSet
--- PASS: TestF5NameSet (0.00s)
=== RUN   TestF5NameList
--- PASS: TestF5NameList (0.00s)
=== RUN   TestValidateEnabledDisabledString
--- PASS: TestValidateEnabledDisabledString (0.00s)
=== RUN   TestValidateReqPrefDisabledString
--- PASS: TestValidateReqPrefDisabledString (0.00s)
PASS
ok      github.com/terraform-providers/terraform-provider-bigip/bigip   65.240s

```

### CODE REVIEW

**Comment by jlosito** [comment]
I rather have the defaults as an empty string.

File: bigip/resource_bigip_ltm_node.go [hunk 1]
```diff
@@ -57,14 +57,19 @@ func resourceBigipLtmNode() *schema.Resource {
...
 				Type:        schema.TypeString,
 				Optional:    true,
+				Default:     "/Common/icmp",
 				Description: "Specifies the name of the monitor or monitor rule that you want to associate with the node.",
-				Computed:    true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "Test-Node",
```
Comment (dannyk81) [inline bigip/resource_bigip_ltm_node.go]: @RavinderReddyF5 this Default shouldn't be here, every user using the latest version of the provider (v0.12.4) now has this string as a description on all their nodes!

/cc @scshitole this needs to be fixed and I would suggest the release to be reverted until then.

File: bigip/resource_bigip_ltm_pool.go [hunk 2]
```diff
@@ -50,6 +50,11 @@ func resourceBigipLtmPool() *schema.Resource {
 				Computed:    true,
 				Description: "Allow SNAT",
 			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "Test-Pool",
```
Comment (dannyk81) [inline bigip/resource_bigip_ltm_pool.go]: and here too

File: bigip/resource_bigip_ltm_virtual_server.go [hunk 3]
```diff
@@ -41,6 +41,11 @@ func resourceBigipLtmVirtualServer() *schema.Resource {
 				Default:     "0.0.0.0/0",
 				Description: "Source IP and mask for the virtual server",
 			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "VirtualServer-test",
```
Comment (dannyk81) [inline bigip/resource_bigip_ltm_virtual_server.go]: and this too

**Comment by dannyk81** [comment]
> I rather have the defaults as an empty string.

I would even suggest removing `Default` entirely, since this is an Optional attribute with user defined description, we should only set it if the user provides a value, otherwise leave it as-is (which is an empty string by default)

**Comment by RavinderReddyF5** [comment]
@dannyk81 
Sure will provide fix.

**Pull request merged**